### PR TITLE
Groups view

### PR DESCRIPTION
**Pull request by justrossthings**
view user's joined groups
browse all groups
filters/search for both

### CODE REVIEW

File: web/src/screens/private/Groups/ViewGroups.js [hunk 1]
```diff
@@ -3,25 +3,173 @@ import PropTypes from 'prop-types';
...
+    const { classes, loading, user } = this.props;
+    const { tab, locationFilter, capabilityFilter, searchFilter } = this.state;
+
+    if (loading) {
+      return <CircularProgress className={classes.progress} size={50} />;
+    }
+
+    const allGroups = user.organisation.groups;
+    const myGroups = user.groups;
+    const allLocations = _.keys(_.transform(allGroups, (result, group) => {
```
Comment (sdunster) [inline web/src/screens/private/Groups/ViewGroups.js]: this logic is pretty hard to follow - it'd be good to toss a comment in explaining the result for future readers
```
// get all tags of type `orgStructure` from all groups in the organisation
```
(even better would be to make a function that does this)

File: web/src/screens/private/Groups/ViewGroups.js [hunk 2]
```diff
@@ -3,25 +3,173 @@ import PropTypes from 'prop-types';
...
+      _.forEach(group.tags, (tag) => {
+        if (tag.type === 'orgStructure') {
+          result[tag.name] = true;
+        }
+      });
+    }, {}));
+    const allCapabilities = _.keys(_.transform(allGroups, (result, group) => {
```
Comment (sdunster) [inline web/src/screens/private/Groups/ViewGroups.js]: as above

**Comment by justrossthings** [comment]
Screenshot:

pasted_image_5_15_18__10_14_am

File: web/src/graphql/current-user.query.js [hunk 3]
```diff
@@ -8,11 +8,28 @@ export default gql`
       displayName
       email
       username
+      organisation {
+        groups {
```
[the remark this answers is not in the document]
Comment (sjmcculloch) [reply web/src/graphql/current-user.query.js]: I mentioned this also @sdunster - we'll have to do a bit of an audit on who uses what for the refetches.

**Comment by sjmcculloch** [comment]
I can help with that styling @justrossthings once it's merged.
